docs/development/pigs/pig-004.rst

```diff
@@ -0,0 +1,131 @@
+700`_, `GH 431`_, `GH 405`_, `GH 228`_ (probably missed a few).
+
+
+Proposal
+========
+
+This proposal is limited. It outlines a few key changes in the setup for
+notebooks and example data that mostly solve the versioning and shipping issue.
+It does not outline e.g. complete lists of notebooks and datasets, those will be
+iteratively improved in the coming month, with or without an extra PIG.
+
+To solve the versioning issue for notebooks, we propose to move the notebooks
+from ``gammapy-extra/notebooks`` to ``gammapy/notebooks``. Whether to store
+notebooks with or without output cells filled in the repository is a difficult
+decision. Removing output cells before committing has the advantage that the
+files are small, and that the diff in the pull request is also small and review
+becomes possible. On the other hand, the output is not directly visible on
+Github. Note that in any case, a rendered version of the notebooks will be
+available via the docs, that is already in place. We propose to remove the
+output and try that workflow for a while, see if we like it. Since notebooks
+aren't very large, and updates are not needed very often, we think changing to
+notebooks with output would also be acceptable in the ``gammapy`` repository.
+
+To solve the versioning issue for data files, we propose to only use stable
+example datasets. Examples are `gammapy-fermi-lat-data`_ or
+`gammapy-extra/datasets/cta-1dc`_ or the upcoming `HESS DL3 DR1`_ or
+``joint-crab`` datasets. Datasets can be in ``gammapy-extra`` or at any other
+URL, but even if they are in ``gammapy-extra``, they should not be ''live''
+datasets. If an issue is found or something is improved, a separate new dataset
+should be added, instead of changing the existing one. So versioning of example
+datasets is not needed.
+
+To ship notebooks and example data to users, we propose to introduce a ``gammapy
+download`` command. This work and discussion how it should work in detail has
+started in `GH 1369`_. Roughly, the idea is that users will use ``gammapy
+download`` to download a version of the notebooks matching the version of the
+Gammapy code, by fetching the files from Github. One question is where to put
+the files on the user machine. The default could be to put the files in
+``$HOME/gammapy-tutorials`` or in ``$CWD/gammapy-tutorial``. The management of
+updating ``gammapy-tutorials`` after a local update of ``gammapy`` is left up to
+the user. We could probably implement a function with some check and version
+printout that we put in the first cell at the top of each notebook. One issue
+with ``gammapy download`` is that one cannot go back and fix or change something
+for released version. E.g. if Github disappears, ``gammapy download`` from older
+versions will no longer work. For this specific issue, a level of indirection
+like a ``gammapy.org/config.json`` file might help. As you can see, this part of
+the proposal is just a draft, feedback welcome!
+
```

Comment:
I am not sure I see the point.
If notebooks are moved to ``gammapy`` to avoid versioning issues, it would be counter-productive to rely on an additional ``gammapy download``command. The later could be a practical way to retrieve ``gammapy-extra`` though.

Reply:
Note that `gammapy.version` always knows it's version. So with one line you can make the URL to fetch the right file from Github or `docs.gammapy.org`. So introducing `gammapy download` does resolve the version coupling issue.

As @joleroi said, there is the option to ship the notebooks with Gammapy. But then probably there you want the rendered version with output filled for users. So when would that rendering happen? IMO it's easier if it happens as part of the docs build and then notebooks are fetched from `docs.gammapy.org`. But it's an open question, more comments on this welcome.